File: SLIDES-markdown/m0/readable.py
"""Chronos: find the positions that are too big.

Run:
    uv run --project ../CODE-ALONGS \
        python m0/readable.py
"""

holdings = [
    {"symbol": "AAPL", "weight": 0.52},
    {"symbol": "MSFT", "weight": 0.31},
    {"symbol": "GLD", "weight": 0.17},
]

for holding in holdings:
    if holding["weight"] > 0.35:
        print(f"Concentrated: {holding['symbol']}")

File: SLIDES-markdown/m0/readable.java
// The same intent, in Java.
import java.util.*;

public class Concentration {
  record Holding(String symbol, double weight) {}

  public static void main(String[] args) {
    List<Holding> holdings = List.of(
      new Holding("AAPL", 0.52),
      new Holding("MSFT", 0.31),
      new Holding("GLD", 0.17)
    );

    for (Holding holding : holdings) {
      if (holding.weight() > 0.35) {
        System.out.println(
          "Concentrated: " + holding.symbol()
        );
      }
    }
  }
}
